lib/html_parser.dart

```diff
@@ -288,7 +304,28 @@ class HtmlParser extends StatelessWidget {
       tree: tree,
       style: context.style.copyOnlyInherited(tree.style),
     );
-
+    List<int> indices = [];
+    tree.children.forEachIndexed((index, element) {
+      //we want the element to be a block element, but we don't want to add
+      //new-lines before/after the html and body
+      if (element.style.display == Display.BLOCK
+          && element.element?.localName != "html"
+          && element.element?.localName != "body"
+      ) {
+        //if the parent element is body and the element is first, we don't want
+        //to add a new-line before
+        if (index == 0 && element.element?.parent?.localName == "body") {
+          indices.add(index + 1);
+        } else {
+          indices.addAll([index, index + 1]);
+        }
+      }
+    });
```

Comment:
This code confuses me. Why do we not detime if a line ending should be added (and actually do that) lower in the code, where we actually return the TextSpan (lines 359-368)? What is the value in predetermining all the locations where to add a line ending?

Reply:
So I see your point, basically this PR copy pasted some of the work I did in #679. 

In that PR, I use the exact same code to determine the indices, but it is used in *two* different places in `parseTree`, so I just predetermined the locations to avoid duplicate code.

I just kept the same code structure here so stuff would merge easier, but I can move it down to the `TextSpan` for this PR if you want.

Reply:
@erickok I merged this PR with your shrinkwrap fixes, so this is now resolved (uses the same API style).